public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx

```diff
@@ -27,123 +28,131 @@ const RECEIVER_STATUS_POLLING_INTERVAL = 10 * 1000; // 10 seconds
-  const fetchReceiverMetadata = alertmanagerApi.endpoints.grafanaNotifiers.useQuery(undefined, {
+  const fetchReceiverMetadata = useGrafanaNotifiersQuery(undefined, {
     skip: !isGrafanaAlertmanager,
   });
 
   // if the OnCall plugin is installed, fetch its list of integrations so we can match those to the Grafana Managed contact points
-  const { data: onCallIntegrations, isLoading: onCallPluginIntegrationsLoading } =
-    onCallApi.endpoints.grafanaOnCallIntegrations.useQuery(undefined, {
-      skip: !onCallPluginInstalled || !isGrafanaAlertmanager,
-    });
-
-  // null = no installed, undefined = loading, [n] is installed with integrations
-  let onCallMetadata: null | undefined | OnCallIntegrationDTO[] = undefined;
-  if (onCallPluginInstalled) {
-    onCallMetadata = onCallIntegrations ?? [];
-  } else if (onCallPluginInstalled === false) {
-    onCallMetadata = null;
-  }
+  const { data: onCallMetadata, isLoading: onCallPluginIntegrationsLoading } = useOnCallIntegrations({
+    skip: !isGrafanaAlertmanager,
+  });
 
   // fetch the latest config from the Alertmanager
   // we use this endpoint only when we need to get the number of policies
-  const fetchAlertmanagerConfiguration = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(
-    selectedAlertmanager!,
-    {
-      refetchOnFocus: true,
-      refetchOnReconnect: true,
-      selectFromResult: (result) => ({
-        ...result,
-        contactPoints: result.data
-          ? enhanceContactPointsWithMetadata(
-              fetchContactPointsStatus.data,
-              fetchReceiverMetadata.data,
-              onCallMetadata,
-              result.data.alertmanager_config.receivers ?? [],
-              result.data
-            )
-          : [],
-      }),
-      skip: !includePoliciesCount,
-    }
-  );
-
-  // for Grafana Managed Alertmanager, we use the new read-only endpoint for getting the list of contact points
-  const fetchGrafanaContactPoints = alertmanagerApi.endpoints.getContactPointsList.useQuery(undefined, {
-    refetchOnFocus: true,
-    refetchOnReconnect: true,
+  const fetchAlertmanagerConfiguration = useGetAlertmanagerConfigurationQuery(selectedAlertmanager!, {
+    ...defaultOptions,
     selectFromResult: (result) => ({
       ...result,
       contactPoints: result.data
         ? enhanceContactPointsWithMetadata(
             fetchContactPointsStatus.data,
             fetchReceiverMetadata.data,
             onCallMetadata,
-            result.data, // contact points from the new readonly endpoint
-            undefined //no config data
+            result.data.alertmanager_config.receivers ?? [],
+            result.data
           )
         : [],
     }),
-    skip: includePoliciesCount || !isGrafanaAlertmanager,
   });
 
   // we will fail silently for fetching OnCall plugin status and integrations
-  const error =
-    fetchAlertmanagerConfiguration.error || fetchGrafanaContactPoints.error || fetchContactPointsStatus.error;
```

Comment:
We only needed `fetchGrafanaContactPoints` when using the response in the alert rule form